frontend/src/pages/ExecutionList.tsx

```diff
@@ -117,25 +117,51 @@ class ExecutionList extends Page<{}, ExecutionListState> {
   }
 
   private async reload(request: ListRequest): Promise<string> {
-    const {
-      error: err,
-      response: res,
-    } = await Apis.getMetadataServicePromiseClient().getExecutions(new GetExecutionsRequest());
-
-    if (err) {
-      // Code === 5 means no record found in backend. This is a temporary workaround.
-      // TODO: remove err.code !== 5 check when backend is fixed.
-      if (err.code !== 5) {
-        this.showPageError(serviceErrorToString(err));
-      }
-      return '';
+    // TODO: Consider making an Api method for returning and caching types
+    if (!this.executionTypesMap || !this.executionTypesMap.size) {
+      this.executionTypesMap = await this.getExecutionTypes();
     }
-
-    const executions = (res && res.getExecutionsList()) || [];
-    await this.getRowsFromExecutions(request, executions);
+    if (!this.state.executions.length) {
+      const executions = await this.getExecutions();
+      this.setState({ executions });
+      this.clearBanner();
+    }
+    this.setState({
+      rows: this.getRowsFromExecutions(request),
+    });
     return '';
   }
 
+  private async getExecutions(): Promise<Execution[]> {
+    const response = await this.api.metadataStoreService.getExecutions(new GetExecutionsRequest());
+
+    if (!response) {
+      this.showPageError('Unable to retrieve Executions.');
+      return [];
+    }
+
+    return response!.getExecutionsList() || [];
+  }
+
+  private async getExecutionTypes(): Promise<Map<number, ExecutionType>> {
+    const response = await this.api.metadataStoreService.getExecutionTypes(
+      new GetExecutionTypesRequest(),
+    );
+
+    if (!response) {
+      this.showPageError('Unable to retrieve Execution Types, some features may not work.');
+      return new Map();
+    }
+
+    const executionTypesMap = new Map<number, ExecutionType>();
+
+    (response!.getExecutionTypesList() || []).forEach(executionType => {
```

Comment:
ditto: `!` won't be needed with `@types/google-protbuf`

Reply:
Done